_posts/2021-09-23-reassign-partitions.md

```diff
@@ -0,0 +1,282 @@
+---
+layout: post
+title:  "Reassign partitions in Kafka Cluster"
+date: 2021-09-23
+author: shubham_rawat
+---
+
+Scaling Kafka Brokers, removing old brokers, and increasing the number of topics and partitions are all pretty common tasks for Kafka users.
+All these cases can cause the load on the brokers to get unbalanced. The most convenient way to manage an unbalanced cluster is to reassign the partitions among all the brokers.
+In Strimzi, this is easily done with the help of the partition reassignment tool: `kafka-reassign-partitions.sh`.
+
+<!--more-->
+
+## Partition Reassignment tool
+
+Within a broker pod, the `kafka-reassign-partitions.sh` tool allows you to reassign partitions to different brokers.
+
+It has three different modes:
+
+| Mode | Description |
+| :-----------: | ------------- |
+| `generate`  | Takes a set of topics and brokers and generates a reassignment JSON file which will result in the partitions of those                     topics being assigned to those brokers. Because this operates on whole topics, it cannot be used when you only want to reassign some partitions for some topics.|
+| `execute` | Takes a reassignment JSON file and applies it to the partitions and brokers in the cluster. Brokers that gain partitions                 as a result become followers of the partition leader. For a given partition, once the new broker has caught up and joined the ISR (in-sync replicas) the old broker will stop being a follower and will delete its replica. |
+| `verify`  | Using the same reassignment JSON file as the `--execute` step, `--verify` checks whether all the partitions in the file have been moved to their intended brokers. If the reassignment is complete, `--verify` also removes any traffic throttles (`--throttle`) that are in effect. Unless removed, throttles will continue to affect the cluster even after the reassignment has finished. |
+
+While using the tool, you have to provide it with two essential JSON files: `topics.json` and `reassignment.json`.
+Wondering what these two JSON files really do? Let's talk a bit about them:
+
+- The `topic.json` basically consists of the topics that we want to reassign or move. Based on this JSON file, our tool will generate a proposal `reassignment.json` which we can further use.
+  
+- The `reassignment.json` file is also generated by the user, but the content for it is generated by the tool, after we feed the tool with the `topic.json` file.
+  We can save that proposal JSON and use it in the JSON file.
+
+## Why use the Partition Reassignment tool?
+
+The Partition Reassignment tool can help you address a variety of use cases.
+
+Some of these are listed here :-
+
+1. There might be cases when you want to shutdown/remove your old broker pod from the cluster and start with some new broker pod.
+   In this scenario, the new pod won't get the partitions assigned to it. So you will have to reassign the
+   partitions among all the brokers.
```

Comment:
This is not how it works with Strimzi. You can scale up the cluster in which a new broker with the next-in-line index number will be added or scale it down in which case the highest index broker will be removed. A Strimzi blog post should reflect how Strimzi works. And not refer to removing old and adding new brokers which is not really possible.